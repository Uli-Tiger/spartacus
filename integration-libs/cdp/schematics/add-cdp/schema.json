{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDPSchematics",
  "title": "CDP Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the CDP feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["CDP"]
    }
  },
  "required": []
}
