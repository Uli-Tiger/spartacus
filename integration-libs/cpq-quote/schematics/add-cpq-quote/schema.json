{
  "$schema": "http://json-schema.org/schema",
  "$id": "CpqQuoteSchematics",
  "title": "cpq-quote Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the cpq-quote.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["cpq-quote"]
    }
  },
  "required": []
}
