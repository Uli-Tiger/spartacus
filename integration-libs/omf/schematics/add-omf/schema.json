{
  "$schema": "http://json-schema.org/schema",
  "$id": "OmfSchematics",
  "title": "OMF Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["OMF"]
    }
  },
  "required": []
}
