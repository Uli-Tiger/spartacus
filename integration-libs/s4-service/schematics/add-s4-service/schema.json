{
  "$schema": "http://json-schema.org/schema",
  "$id": "S4ServiceSchematics",
  "title": "S/4HANA Service Integration Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the S/4HANA Service Integration feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["s4-service"]
    }
  },
  "required": []
}
