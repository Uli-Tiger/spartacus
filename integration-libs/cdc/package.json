{
  "name": "@spartacus/cdc",
  "version": "6.8.0",
  "description": "Customer Data Cloud Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "customer data cloud",
    "gigya",
    "cdc"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cdc",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/forms": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@ng-select/ng-select": "^10.0.3",
    "@ngrx/effects": "^15.3.0",
    "@ngrx/store": "^15.3.0",
    "@spartacus/asm": "6.8.0",
    "@spartacus/core": "6.8.0",
    "@spartacus/organization": "6.8.0",
    "@spartacus/schematics": "6.8.0",
    "@spartacus/storefront": "6.8.0",
    "@spartacus/user": "6.8.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
