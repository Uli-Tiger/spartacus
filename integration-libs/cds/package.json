{
  "name": "@spartacus/cds",
  "version": "6.6.0",
  "description": "Context Driven Service library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "merchandising",
    "segmentation",
    "context-driven services",
    "cds"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/cds",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"src/schematics/**/*.js\" \"src/schematics/**/*.js.map\" \"src/schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@ngrx/store": "^15.3.0",
    "@spartacus/cart": "6.6.0",
    "@spartacus/core": "6.6.0",
    "@spartacus/order": "6.6.0",
    "@spartacus/schematics": "6.6.0",
    "@spartacus/storefront": "6.6.0",
    "@spartacus/tracking": "6.6.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./src/schematics/collection.json"
}
