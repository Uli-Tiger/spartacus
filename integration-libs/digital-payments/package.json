{
  "name": "@spartacus/digital-payments",
  "version": "6.4.0",
  "description": "Digital Payments Integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "digital payments",
    "payments"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/digital-payments",
  "license": "Apache-2.0",
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.4.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.4",
    "@angular/common": "^15.2.4",
    "@angular/core": "^15.2.4",
    "@angular/forms": "^15.2.4",
    "@angular/router": "^15.2.4",
    "@ng-select/ng-select": "^10.0.3",
    "@spartacus/cart": "6.4.0",
    "@spartacus/checkout": "6.4.0",
    "@spartacus/core": "6.4.0",
    "@spartacus/schematics": "6.4.0",
    "@spartacus/storefront": "6.4.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
