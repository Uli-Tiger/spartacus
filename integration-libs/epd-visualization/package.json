{
  "name": "@spartacus/epd-visualization",
  "version": "6.7.0",
  "description": "SAP Enterprise Product Development Visualization integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "epd-visualization",
    "2d",
    "3d",
    "visual",
    "spare",
    "part",
    "picking",
    "epd"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/epd-visualization",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "npm run clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf --glob \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "npm --prefix ../../projects/schematics/ run clean && npm run clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.6.2"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^15.2.9",
    "@angular/common": "^15.2.9",
    "@angular/core": "^15.2.9",
    "@angular/forms": "^15.2.9",
    "@angular/router": "^15.2.9",
    "@sapui5/ts-types-esm": "1.120.1",
    "@spartacus/cart": "6.7.0",
    "@spartacus/core": "6.7.0",
    "@spartacus/schematics": "6.7.0",
    "@spartacus/storefront": "6.7.0",
    "@spartacus/styles": "6.7.0",
    "bootstrap": "^4.6.2",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
