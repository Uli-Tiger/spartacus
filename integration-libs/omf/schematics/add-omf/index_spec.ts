/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  OMF_FEATURE_NAME,
  SPARTACUS_SCHEMATICS,
  SPARTACUS_OMF,
  LibraryOptions as OmfOptions,
  SpartacusOptions,
  omfFeatureModulePath,
  ORDER_FEATURE_NAME,
  orderWrapperModulePath,
  SPARTACUS_ORDER,
  orderFeatureModulePath,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus OMF Schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_OMF,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
    standalone: false,
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: OmfOptions = {
    project: 'schematics-test',
    lazy: false,
    features: [],
  };

  const orderFeatureOptions: OmfOptions = {
    ...libraryNoFeaturesOptions,
    features: [ORDER_FEATURE_NAME],
  };

  const omfFeatureOptions: OmfOptions = {
    ...libraryNoFeaturesOptions,
    features: [OMF_FEATURE_NAME],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );
    schematicRunner.registerCollection(
      SPARTACUS_ORDER,
      path.join(
        __dirname,
        '../../../../feature-libs/order/schematics/collection.json'
      )
    );
    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree
    );

    appTree = await schematicRunner.runExternalSchematic(
      SPARTACUS_SCHEMATICS,
      'ng-add',
      { ...spartacusDefaultOptions, name: 'schematics-test' },
      appTree
    );
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner.runSchematic(
        'ng-add',
        libraryNoFeaturesOptions,
        appTree
      );
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(omfFeatureModulePath)).toBeFalsy();
    });
  });

  describe('OMF feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          orderFeatureOptions,
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          omfFeatureOptions,
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          omfFeatureOptions,
          appTree
        );
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in peerDependencies) {
          if (
            !peerDependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
        }
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(omfFeatureModulePath);
        expect(module).toMatchSnapshot();
      });
      it('should install the appropriate dependencies', async () => {
        const orderWrapperModule = appTree.readContent(orderWrapperModulePath);
        expect(orderWrapperModule).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...orderFeatureOptions, lazy: false },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...omfFeatureOptions, lazy: false },
          appTree
        );
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(omfFeatureModulePath);
        expect(module).toMatchSnapshot();
        expect(appTree.readContent(orderFeatureModulePath)).toBeTruthy();
      });
    });
  });
});
