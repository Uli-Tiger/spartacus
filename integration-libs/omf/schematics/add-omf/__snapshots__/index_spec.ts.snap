// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus OMF Schematics: ng-add OMF feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { OmfRootModule } from "@spartacus/omf/root";

@NgModule({
  declarations: [],
  imports: [
    OmfRootModule
  ]
})
export class OmfFeatureModule { }
"
`;

exports[`Spartacus OMF Schematics: ng-add OMF feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { OmfRootModule } from "@spartacus/omf/root";

@NgModule({
  declarations: [],
  imports: [
    OmfRootModule
  ]
})
export class OmfFeatureModule { }
"
`;

exports[`Spartacus OMF Schematics: ng-add OMF feature general setup should install the appropriate dependencies 1`] = `""`;
